drivers/pwm/pwm_ite_it8xxx2.c

```diff
@@ -162,48 +169,55 @@ static int pwm_it8xxx2_set_cycles(const struct device *dev,
 	 *          CTRx[7:0] value 00h results in a divisor 1
 	 *          CTRx[7:0] value FFh results in a divisor 256
 	 */
-	for (ctr = 0xFF; ctr >= PWM_CTRX_MIN; ctr--) {
-		cxcprs = (((uint32_t) pwm_clk_src) / (ctr + 1) / target_freq);
-		/*
-		 * Make sure cxcprs isn't zero, or we will have
-		 * divide-by-zero on calculating actual_freq.
-		 */
-		if (cxcprs != 0) {
-			actual_freq = ((uint32_t) pwm_clk_src) / (ctr + 1) / cxcprs;
-			if (abs(actual_freq - target_freq) < deviation) {
-				/* CxCPRS[15:0] = cxcprs - 1 */
-				cxcprs--;
-				break;
+	if (target_freq != data->target_freq_prev) {
+		for (data->ctr = 0xFF; data->ctr >= PWM_CTRX_MIN; data->ctr--) {
+			data->cxcprs = (((uint32_t) pwm_clk_src) / (data->ctr + 1) /
```

Comment:
This updates the static data, but there are error paths that return without reprogramming PWM registers.

You should only update the fields in `dev->data` after the frequency change is successful.

Reply:
Done.